### DISCUSSION TITLE
Pls help with fetch

### DISCUSSION THREAD

**Discussion by Potiekhin** [mobxjs/mobx#2994] [General] [answered]
![mobx](https://user-images.githubusercontent.com/49687041/122639577-45bc2f80-d103-11eb-8fae-fe29c3aee1ff.png)
import { makeAutoObservable } from 'mobx'

class Users {
    users = []

    constructor() {
        makeAutoObservable(this)
    }

    getUsers() {
        fetch('https://jsonplaceholder.typicode.com/users')
            .then(response => response.json())
            .then(json => {
                this.users = [...this.users, ...json]
            })
    }
}

export default new Users()
![mobx](https://user-images.githubusercontent.com/49687041/122639561-363ce680-d103-11eb-9b0c-c92ab9b8fe6a.png)

**Comment by kubk** [comment, marked as answer]
You are losing `this` in your example. It looks like you are calling `getUsers` like this:
```js
<Button onClick={userStore.getUsers}/>
```

This won't work because of the JS nature. There are few ways to fix it:

1. Prevent losing `this`:
```js
<Button onClick={() => userStore.getUsers()}/>
```

2. Convert your method to arrow function:

```js
<Button onClick={userStore.getUsers}/>

...

class UserStore {
  ...
  getUsers = () => {
    ...
  }
}
```

3. Use Mobx autobind functionality:

```js
<Button onClick={userStore.getUsers}/>

...

class UserStore {
  ...
  constructor() {
    makeAutoObservable(this, {}, { autoBind: true })
  }
  
  getUsers() {
    ...
  }
}
```

**Reply by Potiekhin** [reply]
Thank you) with autoBind works great)
